### PR TITLE
Consolidate <NativeAotSupported definitions

### PR DESCRIPTION
**Pull request by am11**

### CODE REVIEW

File: src/native/managed/compile-native.proj [hunk 1]
```diff
@@ -13,26 +13,11 @@
         <NativeLibsProjectsToBuild Include="$(MSBuildThisFileDirectory)cdacreader/src/cdacreader.csproj" />
     </ItemGroup>
 
-    <!-- Decide if we're going to do the NativeAOT builds -->
-    <PropertyGroup>
-        <!-- disable on Mono, for now -->
-        <SupportsNativeAotComponents Condition="'$(SupportsNativeAotComponents)' == '' and '$(RuntimeFlavor)' == 'Mono'">false</SupportsNativeAotComponents>
```
Comment (MichalStrehovsky) [inline src/native/managed/compile-native.proj:19]: Cc @lambdageek - we might be losing this exclusion, but that doesn't seem harmful on its own.

File: src/installer/pkg/projects/netcoreappRIDs.props [hunk 2]
```diff
@@ -75,5 +75,11 @@
     <UnofficialBuildRID Include="linux-musl-riscv64">
       <Platform>riscv64</Platform>
     </UnofficialBuildRID>
+    <UnofficialBuildRID Include="linux-loongarch64">
+      <Platform>loongarch64</Platform>
+    </UnofficialBuildRID>
+    <UnofficialBuildRID Include="linux-musl-loongarch64">
+      <Platform>loongarch64</Platform>
+    </UnofficialBuildRID>
```
Comment (MichalStrehovsky) [inline src/installer/pkg/projects/netcoreappRIDs.props:83]:
```suggestion
```

**Review by MichalStrehovsky — APPROVED** [review]
Looks good otherwise, thanks!

**Comment by MichalStrehovsky** [comment]
> I have subsequent changes on top of this to consolidate LocateNativeCompiler targets (scattered in multiple places). When this is merged, I'll open a next PR.

I want to give @lambdageek at least a chance to see this since I'm not directly involved in compile-native and might not know about all the workflows (I don't know why it excludes runtimeflavor mono).

File: src/native/managed/compile-native.proj (hunk 1 above)
Comment (lambdageek) [reply src/native/managed/compile-native.proj:19]: yea it's ok.  I don't think there's anything that depends on `compile-native.proj` when we're building Mono, so this check was probably redundant

**Review by lambdageek — APPROVED** [review]

File: src/native/managed/compile-native.proj (hunk 1 above)
Comment (lambdageek) [reply src/native/managed/compile-native.proj:19]: I think in an earlier iteration of the cdacreader build infrastructure I had compile-native as a separate subset and it was getting pulled in from some mono cross-compiler scenarios.  But now it's only pulled in from coreclr's runtime-prereqs.proj, which shouldn't be triggered by any mono builds

**Comment by am11** [comment]
Failures are unrelated.

**Pull request merged**